model/lightning.py

```diff
@@ -22,27 +22,17 @@ class LightModel(pl.LightningModule):
     criterion: nn.Module
     emb: nn.Sequential
 
-    @staticmethod
-    def add_model_specific_args(parent_parser: ArgumentParser):
-        parser = parent_parser.add_argument_group('Lightning')
-        parser.add_argument('--q_channels', type=int, default=256)
-        parser.add_argument('--emb_channels', type=int, default=128)
-        parser.add_argument('--memory_segment', type=int)
-        parser.add_argument('--padding_method', type=str,
-                            choices=['same', 'valid'], default='same')
-        return parent_parser
-
     def __init__(self, q_channels: int, emb_channels: int, padding_method: str,
                  memory_segment: int,
                  config_dict: dict = None, **kwargs) -> None:
         super().__init__()
-
+        # converting omegaconf into python dict
+        # I usually don't use save_hyperparameters
+        # since hydra already helps you to save a copy of the copy after each run
+        config_dict = OmegaConf.to_container(config_dict, resolve=True)
         self.save_hyperparameters(config_dict)
-        self.save_hyperparameters(kwargs)
-        self.save_hyperparameters(
-            "q_channels", "padding_method", "emb_channels", "memory_segment")
-
-        if self.hparams.memory_segment is not None:
+        self.save_hyperparameters(kwargs)      
```

Comment:
If all the hyperparams are saved in omegaConf, we can just use one args `config: dict`, remove `save_hyperparameters`, and manually assign those variables we want to save as class members.
`OmegaConf.to_container` could also be moved outside before passing the config to the class constructor.

Reply:
Actually `OmegaConf.to_container` can also be removed if `save_hyperparameters` is not needed. `OmegaConf.to_container` is to convert `OmegaConf` objects to be compatible with `save_hyperparameters`.

Please refer to my latest commit [02802da](https://github.com/yoyololicon/wavenet-like-vocoder/pull/3/commits/02802dae2a6ca798cae1ad86ecbc53c5c3bd1bb5) for the new changes.